### PR TITLE
Tree service for notification configuration and cors

### LINKED ISSUE #418: Use the tree service for BucketNotificationConfiguration
We can use [tree service](https://github.com/nspcc-dev/neofs-node/pull/1324) to handle versions of BucketNotificationConfiguration after #407 will be merged.

Use `bucket-notifications` as `SystemName`

## GetBucketNotificationConfiguration
1. `GetNodeByPath` with flag (to get only last version), `SystemName` (to build path) and with attributes `[OID]`
2. Get object from NeoFS

## PutBucketNotificationConfiguration
1. `GetNodeByPath` using `SystemName` (to build path) and with attributes `[OID]`
2. Put object into NeoFS
3. `Add` to root node with attributes `[SystemName, OID]`
4. Delete objects from NeoFS that corresponds node from step 1
5. `Remove` for all nodes from step 1

### PR DESCRIPTION
**Pull request by masterSplinter01**
Closes #418 #419

### CODE REVIEW

File: internal/neofs/tree.go [hunk 1]
```diff
@@ -20,18 +21,15 @@ type TreeClient struct {
...
 
-type Node struct {
-	id   uint64
-	meta map[string]string
-}
-
 const (
 	versioningEnabledKV = "versioning_enabled"
 	lockConfigurationKV = "lock_configuration"
+	OIDKV               = "OID"
```
Comment (alexvanin) [inline internal/neofs/tree.go:27]: This const better be private as all others.

File: api/layer/neofs/tree_service.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package neofs
```
Comment (KirillovDenis) [inline api/layer/neofs/tree_service.go]: Why did we mode tree interface to `neofs` package? It is used in the `layer` package I think it should be there

File: api/layer/neofs/tree_service.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	// GetSettingsNode retrieves the settings node from the tree service and form data.BucketSettings.
+	//
+	// If node is not found returns ErrNodeNotFound error.
+	GetSettingsNode(context.Context, *cid.ID, string) (*data.BucketSettings, error)
+
+	GetNotificationConfigurationNodes(ctx context.Context, cnrID *cid.ID, latestOnly bool) ([]*oid.ID, []uint64, error)
+	PutNotificationConfigurationNode(ctx context.Context, cnrID *cid.ID, objID *oid.ID) error
+	DeleteNotificationConfigurationNode(ctx context.Context, cnrID *cid.ID, nodeID uint64) error
+
+	GetBucketCORS(ctx context.Context, cnrID *cid.ID, latestOnly bool) ([]*oid.ID, []uint64, error)
```
Comment (KirillovDenis) [inline api/layer/neofs/tree_service.go]: I suppose we should return only `oid.ID` of the current (latest) bucket configuration

File: api/layer/neofs/tree_service.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	PutBucketCORS(ctx context.Context, cnrID *cid.ID, objID *oid.ID) error
```
Comment (KirillovDenis) [inline api/layer/neofs/tree_service.go]: We can return not only `error` but also objects to delete

File: api/layer/neofs/tree_service.go [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+	PutSettingsNode(context.Context, *cid.ID, string, *data.BucketSettings) error
+
+	// GetSettingsNode retrieves the settings node from the tree service and form data.BucketSettings.
+	//
+	// If node is not found returns ErrNodeNotFound error.
+	GetSettingsNode(context.Context, *cid.ID, string) (*data.BucketSettings, error)
+
+	GetNotificationConfigurationNodes(ctx context.Context, cnrID *cid.ID, latestOnly bool) ([]*oid.ID, []uint64, error)
+	PutNotificationConfigurationNode(ctx context.Context, cnrID *cid.ID, objID *oid.ID) error
+	DeleteNotificationConfigurationNode(ctx context.Context, cnrID *cid.ID, nodeID uint64) error
```
Comment (KirillovDenis) [inline api/layer/neofs/tree_service.go]: I suppose we can have more general function to delete node from tree

File: api/layer/tree_service.go [hunk 6]
```diff
@@ -15,9 +16,17 @@ type TreeService interface {
...
+	// If node is not found returns ErrNodeNotFound error.
 	GetSettingsNode(context.Context, *cid.ID, string) (*data.BucketSettings, error)
+
+	GetNotificationConfigurationNodes(ctx context.Context, cnrID *cid.ID, latestOnly bool) ([]*oid.ID, []uint64, error)
+	PutNotificationConfigurationNode(ctx context.Context, cnrID *cid.ID, objID *oid.ID) error
+	DeleteNotificationConfigurationNode(ctx context.Context, cnrID *cid.ID, nodeID uint64) error
+
+	GetBucketCORS(ctx context.Context, cnrID *cid.ID, latestOnly bool) ([]*oid.ID, []uint64, error)
+	PutBucketCORS(ctx context.Context, cnrID *cid.ID, objID *oid.ID) error
+	DeleteBucketCORS(ctx context.Context, cnrID *cid.ID, nodeID uint64) error
```
Comment (KirillovDenis) [inline api/layer/tree_service.go:28]: We can drop argument `nodeID uint64` and delete all nodes

**Review by alexvanin — APPROVED** [review]

**Review by KirillovDenis — APPROVED** [review]

**Pull request merged**